### PR TITLE
Kingdom of Exploathing Part 3: Electric Jubilee

### LINKED ISSUE #329: tower init test, actually stop when you have enough to reach 2nd place
when preparing for the tower init test on 2crs, it went well over 2000 initiative, wasting (admittedly cheap) +init items well after it had reached the needed init to get to 2nd place

Before using each item, it should test if init is high enough, and if yes sign up instead of buffing it further

### PR DESCRIPTION
**Pull request by jaspercb**
[Kingdom of Exploathing]
* War support (if you have lots of space chowder, good for you! Otherwise, um... good luck?)
* In softcore, will pull the war kill-doubling mainhand weapons
* Automatically kill The Invader (As early as day 2, if you have lots of shinies. If it can figure out how to get enough resistance and you have Weapon of the Pastalord and the ability to make or pull a meteorb)
* Distinguish `isGeneralStoreAvailable()` (true in KoE) and `isArmoryAvailable()` (false in KoE)
* Sample 2/484 SC KoE runlog: https://docs.google.com/spreadsheets/d/1YuAx3cAtAVPvGcyQ1MKmzh2W4FLR804yw0KnFS7N-v8

[NS challenge improvements]
* Add `sl_secondPlaceOrBust`, which will abort a NS challenge if it can't get second place. Make sl_ascend a sore loser!
* Don't keep piling on +initiative (or stat, or elemental damage) (closes #329 )
* Add the Mime facial expressions to +stat sources

[beta]
* Always be willing to swap out our equipment to meet our desired combat or noncombat rate. This especially helps in zones where we want to run +combat but previously weren't willing to swap out our equipment to do so, because we couldn't be sure we weren't replacing something important. With the new `slEquip` system, we can be sure we're not replacing something important.

[for my own sanity]
* Set the KoLmafia settings to _print the damn stack when we abort so we get slightly better bug reports_

### CODE REVIEW

File: RELEASE/scripts/sl_ascend.ash [hunk 1]
```diff
@@ -943,6 +943,7 @@ void maximize_hedge()
 	element third = ns_hedge3();
 	if((first == $element[none]) || (second == $element[none]) || (third == $element[none]))
 	{
+		doHottub();
```
Comment (soolar) [inline RELEASE/scripts/sl_ascend.ash]: Should we maybe check if we have flared nostrils before calling `doHottub` so we don't waste a hottub use on nothing if we don't have it?

File: RELEASE/scripts/sl_ascend/sl_combat.ash [hunk 2]
```diff
@@ -583,6 +583,11 @@ string sl_combatHandler(int round, string opp, string text)
 		}
 	}
 
+	if(enemy.to_string() == "the invader" && sl_have_skill($skill[Weapon of the Pastalord]))
+	{
+		return "skill " + $skill[Weapon of the Pastalord];
```
Comment (soolar) [inline RELEASE/scripts/sl_ascend/sl_combat.ash]: `return useSkill($skill[Weapon of the Pastalord], false);` is preferable.

File: RELEASE/scripts/sl_ascend/sl_koe.ash [hunk 3]
```diff
@@ -19,3 +19,65 @@ boolean koe_initializeSettings()
...
+{
+	if(!in_koe())
+	{
+		return false;
+	}
+
+	buffMaintain($effect[Astral Shell], 10, 1, 1);
+	buffMaintain($effect[Elemental Saucesphere], 10, 1, 1);
+	buffMaintain($effect[Hide of Sobek], 10, 1, 1);
+	buffMaintain($effect[Spectral Awareness], 10, 1, 1);
```
Comment (soolar) [inline RELEASE/scripts/sl_ascend/sl_koe.ash]: I don't think we're going to have Hide of Sobek or Spectral Awareness in KoE :P

File: RELEASE/scripts/sl_ascend.ash (hunk 1 above)
Comment (jaspercb) [reply RELEASE/scripts/sl_ascend.ash]: My reasoning here was "there might be other bad effects that are worth removing here", but looking at the wiki that doesn't appear to be true. Will guard with a `have_effect()` check.

File: RELEASE/scripts/sl_ascend/sl_combat.ash (hunk 2 above)
Comment (jaspercb) [reply RELEASE/scripts/sl_ascend/sl_combat.ash]: 👍

File: RELEASE/scripts/sl_ascend/sl_koe.ash (hunk 3 above)
Comment (jaspercb) [reply RELEASE/scripts/sl_ascend/sl_koe.ash]: 😆

**Pull request merged**